torch/storage.py

```diff
@@ -43,7 +43,8 @@ def clone(self):
         """Returns a copy of this storage"""
         device = self.get_device() if self.is_cuda else -1
         with torch.cuda.device(device):
-            return type(self)(self.size()).copy_(self)
+            new_tensor = torch.Tensor(self)
```

Comment:
If I understand well, this will create a float tensor with default tensor type. And, I think, it currently fails if storage dtype does not match exactly that default tensor type

Reply:
Indeed:
```
>>> torch.Tensor(torch.ByteStorage([1,2,3]))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
RuntimeError: result.storage().dtype() == storage.dtype() INTERNAL ASSERT FAILED at /opt/conda/conda-bld/pytorch_1587107261878/work/aten/src/ATen/native/Resize.h:100, please report a bug to PyTorch.
```

Reply:
yeah [it](https://github.com/pytorch/pytorch/issues/37457) was a bug. fixed it

Reply:
`test_new` in `test_torch` fails with my current fix because this 
```
 x = torch.autograd.Variable(torch.Tensor())
 z = torch.autograd.Variable(torch.IntTensor([1, 2, 3]))
 x.new(z.storage())
```
is currently expected to throw a RuntimeError. not sure if we want this behavior or not. anyway, I think [this](https://github.com/pytorch/pytorch/issues/37457) bug should be fixed in a separate PR since it's unrelated to this PR